improc/simulator.py

```diff
@@ -453,6 +501,331 @@ def get_star_flux_values(self):
+        ----------
+        imsize: tuple, default (100, 100)
+            The size of the image (y, x).
+        center_x: float, default None
+            The x position of the galaxy center inside the image.
+            If given as None, will position at the center of the image.
+        center_y: float, default None
+            The y position of the galaxy center inside the image.
+            If given as None, will position at the center of the image.
+        exp_scale: float, default 1.0
+            The length scale used to make the disk.
+        sersic_scale: float, default 1.0
+            The length scale used to make the bulge.
+        exp_flux: float, default 1.0
+            The total flux of the disk. The disk flux is modulated
+            by a factor cos_i, so the total flux of the disk,
+            if it were viewed face-on, is exp_flux, but we only
+            see a fraction of it, depending on the inclination.
+        sersic_flux: float, default 0.01
+            The total flux of the bulge.
+            Note that this is a very sharp profile,
+            so its contribution to the flux is usually small.
+        cos_i: float, default None
+            The inclination angle cosine that determines how flat and narrow
+            the disk of the galaxy will be. One means face-on, zero means edge-on.
+            If not given, will choose a uniform cos_i in range [0.1,1].
+        rotation: float, default None
+            The rotation angle of the galaxy, in degrees.
+            Zero means the galaxy is oriented North-South, and the angle increases
+            towards East. If not given, will choose a uniform rotation in range [0,360].
+
+        Returns
+        -------
+        galaxy_image: np.ndarray
+            The image of the galaxy, normalized to total_flux.
+        """
+        if center_x is None:
+            center_x = imsize[1] // 2
+        if center_y is None:
+            center_y = imsize[0] // 2
+        if cos_i is None:
+            cos_i = np.random.uniform(0.1, 1)
+        if rotation is None:
+            rotation = np.random.uniform(0, 360)
+
+        rotation = np.deg2rad(rotation)
+
+        # regular coordinate grid
+        x0, y0 = np.meshgrid(np.arange(imsize[1]), np.arange(imsize[0]))
+
+        # transform the coordinates using rotation and translation
+        x = (x0 - center_x) * np.cos(rotation) + (y0 - center_y) * np.sin(rotation)
+        y = (y0 - center_y) * np.cos(rotation) - (x0 - center_x) * np.sin(rotation)
+        r = np.sqrt(x ** 2 + y ** 2)
+
+        # first make the bulge using a sersic profile
+        r0 = sersic_scale
+        bulge = np.exp(-7.67 * (r / r0) ** (1 / 4))
+        bulge *= sersic_flux / np.sum(bulge)
+
+        # now make the disk
+        r0 = exp_scale
+        disk = np.exp(-1.67 * np.sqrt((x / r0 / cos_i) ** 2 + (y / r0) ** 2))
+        disk *= exp_flux / np.sum(disk)
+        disk *= cos_i  # correct for the projection of the disk
```

Comment:
I don't understand why this is here.  Ignoring the effects of dust, galaxies don't have directed radiation.  The total flux should be the same in any event-- there are the same number of stars radiating, no matter how the galaxy is oriented relative to us.  An inclined galaxy would just have a higher surface brightness.

Reply:
I was thinking about galaxies as optically thick objects, but I realize they are not. 
The reason I put in this cos_i term (after a lot of deliberation) is that if you put the same total flux for two galaxies, one edge-on and one face-on, the face-on one will have a much lower surface brightness, which I wanted to avoid. 

Now if you convince me that galaxies are total transparent to their own starlight, then indeed we should remove this term. But can we actually totally ignore the dust? I know this is getting really low in the weeds for such a simple simulator, but my intuition is that edge-on galaxies are not that much brighter than face on, but maybe I need to go look at more deep field images or something...